rust-runtime/smithy-eventstream/src/frame.rs

```diff
@@ -19,6 +20,29 @@ const MESSAGE_CRC_LENGTH_BYTES: u32 = size_of::<u32>() as u32;
 const MAX_HEADER_NAME_LEN: usize = 255;
 const MIN_HEADER_LEN: usize = 2;
 
+pub type SignMessageError = Box<dyn StdError + Send + Sync + 'static>;
+
+/// Signs an Event Stream message.
+pub trait SignMessage {
```

Comment:
should probably eventually be able to remove this trait since there's really only one signing implementation

Reply:
I was working around crate boundaries with this. None of the Smithy crates know about the signers, and that's where all of this code is being placed right now.

Also, since Event Stream is a generic Smithy construct and not tied to the AWS protocols, the signing implementation should be configurable.